File: src/main/java/io/github/gk0wk/violet/database/DefaultProperties.java
package io.github.gk0wk.violet.database;

import com.google.common.collect.ImmutableMap;

import java.util.Map;
import java.util.concurrent.TimeUnit;

public class DefaultProperties {
    static final Map<String, String> MySQL = ImmutableMap.<String, String> builder()
            // Set timezone
            .put("serverTimezone", "Asia/Shanghai")
            // Ensure we use utf8 encoding
            .put("useUnicode", "true")
            .put("characterEncoding", "utf8")
            // https://github.com/brettwooldridge/HikariCP/wiki/MySQL-Configuration
            .put("cachePrepStmts", "true")
            .put("prepStmtCacheSize", "250")
            .put("prepStmtCacheSqlLimit", "2048")
            .put("useServerPrepStmts", "true")
            .put("useLocalSessionState", "true")
            .put("rewriteBatchedStatements", "true")
            .put("cacheResultSetMetadata", "true")
            .put("cacheServerConfiguration", "true")
            .put("elideSetAutoCommits", "true")
            .put("maintainTimeStats", "false")
            .put("alwaysSendSetIsolation", "false")
            .put("cacheCallableStmts", "true")
            // Set the driver level TCP socket timeout
            // See: https://github.com/brettwooldridge/HikariCP/wiki/Rapid-Recovery
            .put("socketTimeout", String.valueOf(TimeUnit.SECONDS.toMillis(30)))
            .build();
}

File: src/main/kotlin/city/newnan/violet/sql/DefaultProperties.kt
package city.newnan.violet.sql

import java.util.concurrent.TimeUnit

object DefaultProperties {
    val MySQL = mapOf(
        // Set timezone
        "serverTimezone" to "Asia/Shanghai",
        // Ensure we use utf8 encoding
        "useUnicode" to "true",
        "characterEncoding" to "utf8",
        // https://github.com/brettwooldridge/HikariCP/wiki/MySQL-Configuration
        "cachePrepStmts" to "true",
        "prepStmtCacheSize" to "250",
        "prepStmtCacheSqlLimit" to "2048",
        "useServerPrepStmts" to "true",
        "useLocalSessionState" to "true",
        "rewriteBatchedStatements" to "true",
        "cacheResultSetMetadata" to "true",
        "cacheServerConfiguration" to "true",
        "elideSetAutoCommits" to "true",
        "maintainTimeStats" to "false",
        "alwaysSendSetIsolation" to "false",
        "cacheCallableStmts" to "true",
        // Set the driver level TCP socket timeout
        // See: https://github.com/brettwooldridge/HikariCP/wiki/Rapid-Recovery
        "socketTimeout" to TimeUnit.SECONDS.toMillis(30).toString()
    )
}
